ui/src/app/views/dataset-page.tsx

```diff
@@ -410,6 +416,34 @@ const DataSetPage = fp.flow(withCurrentWorkspace(), withUrlParams())(
           selectedPreviewDomain: dataSetPreviewResp.domainValue[0].domain
         });
       } catch (ex) {
+        const exceptionResponse = await ex.json() as unknown as ErrorResponse;
+        let errorText: string;
+        if (exceptionResponse.statusCode === 400) {
```

Comment:
will it be neater to have a switch case on statusCode. its just difficult to read